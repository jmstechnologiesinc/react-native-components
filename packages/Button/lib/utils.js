"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getButtonColors = void 0;
var react_native_1 = require("react-native");
var color_1 = __importDefault(require("color"));
var colors_1 = require("./colors");
var LightTheme_1 = __importDefault(require("./LightTheme"));
var isDark = function (_a) {
    var dark = _a.dark, backgroundColor = _a.backgroundColor;
    if (typeof dark === 'boolean') {
        return dark;
    }
    if (backgroundColor === 'transparent') {
        return false;
    }
    if (backgroundColor !== 'transparent') {
        return !(0, color_1.default)(backgroundColor).isLight();
    }
    return false;
};
var getButtonBackgroundColor = function (_a) {
    var isMode = _a.isMode, disabled = _a.disabled, customButtonColor = _a.customButtonColor;
    if (customButtonColor && !disabled) {
        return customButtonColor;
    }
    if (LightTheme_1.default.isV3) {
        if (disabled) {
            if (isMode('outlined') || isMode('text')) {
                return 'transparent';
            }
            return LightTheme_1.default.colors.surfaceDisabled;
        }
        if (isMode('elevated')) {
            return LightTheme_1.default.colors.elevation.level1;
        }
        if (isMode('contained')) {
            return LightTheme_1.default.colors.primary;
        }
        if (isMode('contained-tonal')) {
            return LightTheme_1.default.colors.secondaryContainer;
        }
    }
    if (isMode('contained')) {
        if (disabled) {
            return (0, color_1.default)(LightTheme_1.default.dark ? colors_1.white : colors_1.black)
                .alpha(0.12)
                .rgb()
                .string();
        }
        return LightTheme_1.default.colors.primary;
    }
    return 'transparent';
};
var getButtonTextColor = function (_a) {
    var isMode = _a.isMode, disabled = _a.disabled, customTextColor = _a.customTextColor, backgroundColor = _a.backgroundColor, dark = _a.dark;
    if (customTextColor && !disabled) {
        return customTextColor;
    }
    if (LightTheme_1.default.isV3) {
        if (disabled) {
            return LightTheme_1.default.colors.onSurfaceDisabled;
        }
        if (typeof dark === 'boolean') {
            if (isMode('contained') ||
                isMode('contained-tonal') ||
                isMode('elevated')) {
                return isDark({ dark: dark, backgroundColor: backgroundColor }) ? colors_1.white : colors_1.black;
            }
        }
        if (isMode('outlined') || isMode('text') || isMode('elevated')) {
            return LightTheme_1.default.colors.primary;
        }
        if (isMode('contained')) {
            return LightTheme_1.default.colors.onPrimary;
        }
        if (isMode('contained-tonal')) {
            return LightTheme_1.default.colors.onSecondaryContainer;
        }
    }
    if (disabled) {
        return (0, color_1.default)(LightTheme_1.default.dark ? colors_1.white : colors_1.black)
            .alpha(0.32)
            .rgb()
            .string();
    }
    if (isMode('contained')) {
        return isDark({ dark: dark, backgroundColor: backgroundColor }) ? colors_1.white : colors_1.black;
    }
    return LightTheme_1.default.colors.primary;
};
var getButtonBorderColor = function (_a) {
    var isMode = _a.isMode, disabled = _a.disabled;
    if (LightTheme_1.default.isV3) {
        if (disabled && isMode('outlined')) {
            return LightTheme_1.default.colors.surfaceDisabled;
        }
        if (isMode('outlined')) {
            return LightTheme_1.default.colors.outline;
        }
    }
    if (isMode('outlined')) {
        return (0, color_1.default)(LightTheme_1.default.dark ? colors_1.white : colors_1.black)
            .alpha(0.29)
            .rgb()
            .string();
    }
    return 'transparent';
};
var getButtonBorderWidth = function (_a) {
    var isMode = _a.isMode;
    if (LightTheme_1.default.isV3) {
        if (isMode('outlined')) {
            return 1;
        }
    }
    if (isMode('outlined')) {
        return react_native_1.StyleSheet.hairlineWidth;
    }
    return 0;
};
var getButtonColors = function (_a) {
    var mode = _a.mode, customButtonColor = _a.customButtonColor, customTextColor = _a.customTextColor, disabled = _a.disabled, dark = _a.dark;
    var isMode = function (modeToCompare) {
        return mode === modeToCompare;
    };
    var backgroundColor = getButtonBackgroundColor({
        isMode: isMode,
        disabled: disabled,
        customButtonColor: customButtonColor,
    });
    var textColor = getButtonTextColor({
        isMode: isMode,
        disabled: disabled,
        customTextColor: customTextColor,
        backgroundColor: backgroundColor,
        dark: dark,
    });
    var borderColor = getButtonBorderColor({ isMode: isMode, disabled: disabled });
    var borderWidth = getButtonBorderWidth({ isMode: isMode });
    return {
        backgroundColor: backgroundColor,
        borderColor: borderColor,
        textColor: textColor,
        borderWidth: borderWidth,
    };
};
exports.getButtonColors = getButtonColors;
