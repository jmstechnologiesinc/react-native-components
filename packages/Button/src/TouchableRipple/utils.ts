import color from 'color';
import theme from '../LightTheme';

const getUnderlayColor = ({
  calculatedRippleColor,
  underlayColor,
}: {
  calculatedRippleColor: string;
  underlayColor?: string;
}) => {
  if (underlayColor != null) {
    return underlayColor;
  }

  if (theme.isV3) {
    return color(calculatedRippleColor).rgb().string();
  }

  return color(calculatedRippleColor).fade(0.5).rgb().string();
};

const getRippleColor = ({rippleColor}: {rippleColor?: string}) => {
  if (rippleColor) {
    return rippleColor;
  }

  if (theme.isV3) {
    return color(theme.colors.onSurface).alpha(0.12).rgb().string();
  }

  if (theme.dark) {
    return color(theme.colors.text).alpha(0.32).rgb().string();
  }
  return color(theme.colors.text).alpha(0.2).rgb().string();
};

export const getTouchableRippleColors = ({
  rippleColor,
  underlayColor,
}: {
  rippleColor?: string;
  underlayColor?: string;
}) => {
  const calculatedRippleColor = getRippleColor({rippleColor});
  return {
    calculatedRippleColor,
    calculatedUnderlayColor: getUnderlayColor({
      calculatedRippleColor,
      underlayColor,
    }),
  };
};
