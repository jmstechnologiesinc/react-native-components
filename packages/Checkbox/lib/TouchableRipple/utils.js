"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getTouchableRippleColors = void 0;
var color_1 = __importDefault(require("color"));
var LightTheme_1 = __importDefault(require("../LightTheme"));
var getUnderlayColor = function (_a) {
    var calculatedRippleColor = _a.calculatedRippleColor, underlayColor = _a.underlayColor;
    if (underlayColor != null) {
        return underlayColor;
    }
    if (LightTheme_1.default.isV3) {
        return (0, color_1.default)(calculatedRippleColor).rgb().string();
    }
    return (0, color_1.default)(calculatedRippleColor).fade(0.5).rgb().string();
};
var getRippleColor = function (_a) {
    var rippleColor = _a.rippleColor;
    if (rippleColor) {
        return rippleColor;
    }
    if (LightTheme_1.default.isV3) {
        return (0, color_1.default)(LightTheme_1.default.colors.onSurface).alpha(0.12).rgb().string();
    }
    if (LightTheme_1.default.dark) {
        return (0, color_1.default)(LightTheme_1.default.colors.text).alpha(0.32).rgb().string();
    }
    return (0, color_1.default)(LightTheme_1.default.colors.text).alpha(0.2).rgb().string();
};
var getTouchableRippleColors = function (_a) {
    var rippleColor = _a.rippleColor, underlayColor = _a.underlayColor;
    var calculatedRippleColor = getRippleColor({ rippleColor: rippleColor });
    return {
        calculatedRippleColor: calculatedRippleColor,
        calculatedUnderlayColor: getUnderlayColor({
            calculatedRippleColor: calculatedRippleColor,
            underlayColor: underlayColor,
        }),
    };
};
exports.getTouchableRippleColors = getTouchableRippleColors;
