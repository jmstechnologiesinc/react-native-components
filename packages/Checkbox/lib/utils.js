"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getSelectionControlIOSColor = exports.getAndroidSelectionControlColor = void 0;
var color_1 = __importDefault(require("color"));
var LightTheme_1 = __importDefault(require("./LightTheme"));
var getAndroidCheckedColor = function (_a) {
    var customColor = _a.customColor;
    if (customColor) {
        return customColor;
    }
    if (LightTheme_1.default.isV3) {
        return LightTheme_1.default.colors.primary;
    }
    return LightTheme_1.default.colors.accent;
};
var getAndroidUncheckedColor = function (_a) {
    var customUncheckedColor = _a.customUncheckedColor;
    if (customUncheckedColor) {
        return customUncheckedColor;
    }
    if (LightTheme_1.default.isV3) {
        return LightTheme_1.default.colors.onSurfaceVariant;
    }
    if (LightTheme_1.default.dark) {
        return (0, color_1.default)(LightTheme_1.default.colors.text).alpha(0.7).rgb().string();
    }
    return (0, color_1.default)(LightTheme_1.default.colors.text).alpha(0.54).rgb().string();
};
var getAndroidRippleColor = function (_a) {
    var checkedColor = _a.checkedColor, disabled = _a.disabled;
    if (disabled) {
        if (LightTheme_1.default.isV3) {
            return (0, color_1.default)(LightTheme_1.default.colors.onSurface).alpha(0.16).rgb().string();
        }
        return (0, color_1.default)(LightTheme_1.default.colors.text).alpha(0.16).rgb().string();
    }
    return (0, color_1.default)(checkedColor).fade(0.32).rgb().string();
};
var getAndroidControlColor = function (_a) {
    var checked = _a.checked, disabled = _a.disabled, checkedColor = _a.checkedColor, uncheckedColor = _a.uncheckedColor;
    if (disabled) {
        if (LightTheme_1.default.isV3) {
            return LightTheme_1.default.colors.onSurfaceDisabled;
        }
        return LightTheme_1.default.colors.text;
    }
    if (checked) {
        return checkedColor;
    }
    return uncheckedColor;
};
var getAndroidSelectionControlColor = function (_a) {
    var disabled = _a.disabled, checked = _a.checked, customColor = _a.customColor, customUncheckedColor = _a.customUncheckedColor;
    var checkedColor = getAndroidCheckedColor({ customColor: customColor });
    var uncheckedColor = getAndroidUncheckedColor({
        customUncheckedColor: customUncheckedColor,
    });
    return {
        rippleColor: getAndroidRippleColor({ checkedColor: checkedColor, disabled: disabled }),
        selectionControlColor: getAndroidControlColor({
            disabled: disabled,
            checked: checked,
            checkedColor: checkedColor,
            uncheckedColor: uncheckedColor,
        }),
    };
};
exports.getAndroidSelectionControlColor = getAndroidSelectionControlColor;
var getIOSCheckedColor = function (_a) {
    var disabled = _a.disabled, customColor = _a.customColor;
    if (disabled) {
        if (LightTheme_1.default.isV3) {
            return LightTheme_1.default.colors.onSurfaceDisabled;
        }
        return LightTheme_1.default.colors.disabled;
    }
    if (customColor) {
        return customColor;
    }
    if (LightTheme_1.default.isV3) {
        return LightTheme_1.default.colors.primary;
    }
    return LightTheme_1.default.colors.accent;
};
var getIOSRippleColor = function (_a) {
    var checkedColor = _a.checkedColor, disabled = _a.disabled;
    if (disabled) {
        if (LightTheme_1.default.isV3) {
            return (0, color_1.default)(LightTheme_1.default.colors.onSurface).alpha(0.16).rgb().string();
        }
        return (0, color_1.default)(LightTheme_1.default.colors.text).alpha(0.16).rgb().string();
    }
    return (0, color_1.default)(checkedColor).fade(0.32).rgb().string();
};
var getSelectionControlIOSColor = function (_a) {
    var disabled = _a.disabled, customColor = _a.customColor;
    var checkedColor = getIOSCheckedColor({ disabled: disabled, customColor: customColor });
    return {
        checkedColor: checkedColor,
        rippleColor: getIOSRippleColor({
            checkedColor: checkedColor,
            disabled: disabled,
        }),
    };
};
exports.getSelectionControlIOSColor = getSelectionControlIOSColor;
