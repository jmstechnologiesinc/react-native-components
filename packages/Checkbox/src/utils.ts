import color from 'color';
import theme from './LightTheme';

const getAndroidCheckedColor = ({customColor}: {customColor?: string}) => {
  if (customColor) {
    return customColor;
  }

  if (theme.isV3) {
    return theme.colors.primary;
  }

  return theme.colors.accent;
};

const getAndroidUncheckedColor = ({
  customUncheckedColor,
}: {
  customUncheckedColor?: string;
}) => {
  if (customUncheckedColor) {
    return customUncheckedColor;
  }

  if (theme.isV3) {
    return theme.colors.onSurfaceVariant;
  }

  if (theme.dark) {
    return color(theme.colors.text).alpha(0.7).rgb().string();
  }

  return color(theme.colors.text).alpha(0.54).rgb().string();
};

const getAndroidRippleColor = ({
  checkedColor,
  disabled,
}: {
  checkedColor: string;
  disabled?: boolean;
}) => {
  if (disabled) {
    if (theme.isV3) {
      return color(theme.colors.onSurface).alpha(0.16).rgb().string();
    }
    return color(theme.colors.text).alpha(0.16).rgb().string();
  }

  return color(checkedColor).fade(0.32).rgb().string();
};

const getAndroidControlColor = ({
  checked,
  disabled,
  checkedColor,
  uncheckedColor,
}: {
  checked: boolean;
  checkedColor: string;
  uncheckedColor: string;
  disabled?: boolean;
}) => {
  if (disabled) {
    if (theme.isV3) {
      return theme.colors.onSurfaceDisabled;
    }
    return theme.colors.text;
  }

  if (checked) {
    return checkedColor;
  }
  return uncheckedColor;
};

export const getAndroidSelectionControlColor = ({
  disabled,
  checked,
  customColor,
  customUncheckedColor,
}: {
  checked: boolean;
  disabled?: boolean;
  customColor?: string;
  customUncheckedColor?: string;
}) => {
  const checkedColor = getAndroidCheckedColor({customColor});
  const uncheckedColor = getAndroidUncheckedColor({
    customUncheckedColor,
  });
  return {
    rippleColor: getAndroidRippleColor({checkedColor, disabled}),
    selectionControlColor: getAndroidControlColor({
      disabled,
      checked,
      checkedColor,
      uncheckedColor,
    }),
  };
};

const getIOSCheckedColor = ({
  disabled,
  customColor,
}: {
  customColor?: string;
  disabled?: boolean;
}) => {
  if (disabled) {
    if (theme.isV3) {
      return theme.colors.onSurfaceDisabled;
    }
    return theme.colors.disabled;
  }

  if (customColor) {
    return customColor;
  }

  if (theme.isV3) {
    return theme.colors.primary;
  }

  return theme.colors.accent;
};

const getIOSRippleColor = ({
  checkedColor,
  disabled,
}: {
  checkedColor: string;
  disabled?: boolean;
}) => {
  if (disabled) {
    if (theme.isV3) {
      return color(theme.colors.onSurface).alpha(0.16).rgb().string();
    }
    return color(theme.colors.text).alpha(0.16).rgb().string();
  }
  return color(checkedColor).fade(0.32).rgb().string();
};

export const getSelectionControlIOSColor = ({
  disabled,
  customColor,
}: {
  disabled?: boolean;
  customColor?: string;
}) => {
  const checkedColor = getIOSCheckedColor({disabled, customColor});
  return {
    checkedColor,
    rippleColor: getIOSRippleColor({
      checkedColor,
      disabled,
    }),
  };
};
