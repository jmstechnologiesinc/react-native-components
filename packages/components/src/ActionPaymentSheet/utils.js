export const showActionSheet = (actionSheetRef) => {
    actionSheetRef.current.show();
};

export const hideActionSheet = (actionSheetRef) => {
    actionSheetRef.current.hide();
};

export const onUpdateValue = (paymentOptions, index, callback) => {
    for (let i = 0; i < paymentOptions.length; i++) {
        if (i === index) {
            if (paymentOptions[i].hasOwnProperty('selected')) {
                paymentOptions[i].selected = true;
            }
        } else {
            if (paymentOptions[i].hasOwnProperty('selected')) {
                paymentOptions[i].selected = false;
            }
        }
    }

    return callback([...paymentOptions]);
};
