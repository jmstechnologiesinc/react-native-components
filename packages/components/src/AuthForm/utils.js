 
 import { localized } from '../Localization/Localization';
 import { VENDOR_INDUSTRIES_MAPPING } from '@jmsstudiosinc/vendor';




export const handleDateOfBirhtChange = (text, inputActionHandler, callback, dateOfBirth) => {
        if (
            (text.length === 2 && dateOfBirth.length === 3) ||
            (text.length === 5 && dateOfBirth.length === 6)
        ) {
            text = text.slice(0, -1);
        } else if (text.length === 2 || text.length === 5) {
            text += "/";
        }
        callback(text)    
        return inputActionHandler('dateofBirth', text )
}

 export const OPTIONS = [
        {
            item: localized(VENDOR_INDUSTRIES_MAPPING.Restaurant.title),
            selected: false
        },
        {
            item: localized(VENDOR_INDUSTRIES_MAPPING.GroceryGourmet.title),
            selected: false
        },
        {
            item: localized(VENDOR_INDUSTRIES_MAPPING.Liquor.title),
            selected: false
        },
        {
            item: localized(VENDOR_INDUSTRIES_MAPPING.Pharmacy.title),
            selected: false
        },
          {
            item: localized(VENDOR_INDUSTRIES_MAPPING.VideoGames.title),
            selected: false
        },
          {
            item: localized(VENDOR_INDUSTRIES_MAPPING.LuggageTravelGear.title),
            selected: false
        },
          {
            item: localized(VENDOR_INDUSTRIES_MAPPING.ToysGames.title),
            selected: false
        },
       
        {
            item: localized(VENDOR_INDUSTRIES_MAPPING.GardenOutdoor.title),
            selected: false
        },
        {
            item: localized(VENDOR_INDUSTRIES_MAPPING.Electronics.title),
            selected: false
        },
        {
            item: localized(VENDOR_INDUSTRIES_MAPPING.Computers.title),
            selected: false
        },
        {
            item: localized(VENDOR_INDUSTRIES_MAPPING.CellPhones.title),
            selected: false
        },
        {
            item: localized(VENDOR_INDUSTRIES_MAPPING.Books.title),
            selected: false
        },
        {
            item: localized(VENDOR_INDUSTRIES_MAPPING.BeautyPersonalCare.title),
            selected: false
        },
        {
            item: localized(VENDOR_INDUSTRIES_MAPPING.Baby.title),
            selected: false
        },

    ]