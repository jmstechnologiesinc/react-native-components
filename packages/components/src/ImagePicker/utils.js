import { PermissionsAndroid } from 'react-native';

export const checkAndAskForPermission = () =>
    new Promise(async (resolve, reject) => {
        let permissionStatus;
        try {
            if (Platform.OS === 'android') {
                permissionStatus = await PermissionsAndroid.request(PermissionsAndroid.PERMISSIONS.CAMERA);
                // permissionStatus = await PermissionsAndroid.request(android.permission.WRITE_EXTERNAL_STORAGE);
            } else {
                //  Missing permission settings for ios
                permissionStatus = 'granted';
            }

            if (permissionStatus === 'granted') {
                resolve(permissionStatus);
            } else {
                reject(permissionStatus);
            }
        } catch (e) {
            reject(permissionStatus);
        }
    });

export const IMAGE_PICKER_ACTIONS = {
    launchCamera: 'launchCamera',
    launchImageLibrary: 'launchImageLibrary',
    removeImage: 'removeImage',
    cancel: 'cancel',
};

export const options = [
    { title: 'Take photo', value: IMAGE_PICKER_ACTIONS.launchCamera, icon: 'camera' },
    { title: 'Choose from library', value: IMAGE_PICKER_ACTIONS.launchImageLibrary, icon: 'folder-image' },
    { title: 'Remove Profile Photo', value: IMAGE_PICKER_ACTIONS.removeImage, icon: 'image-remove' },
    { title: 'cancel', value: IMAGE_PICKER_ACTIONS.cancel, icon: 'cancel' },
];
