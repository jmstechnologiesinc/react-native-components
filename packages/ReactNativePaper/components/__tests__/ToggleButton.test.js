import * as React from 'react';
import renderer from 'react-test-renderer';
import color from 'color';
import ToggleButton from '../ToggleButton';
import { getToggleButtonColor } from '../ToggleButton/utils';
import { tokens } from '../../styles/themes/v3/tokens';
import MD3LightTheme from '../../styles/themes/v3/LightTheme';
import MD2LightTheme from '../../styles/themes/v2/LightTheme';
import MD3DarkTheme from '../../styles/themes/v3/DarkTheme';
import MD2DarkTheme from '../../styles/themes/v2/DarkTheme';

it('renders toggle button', () => {
  const tree = renderer
    .create(<ToggleButton status="checked" onPress={() => {}} icon="heart" />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders disabled toggle button', () => {
  const tree = renderer
    .create(
      <ToggleButton
        disabled
        value="toggle"
        status="checked"
        onValueChange={() => {}}
        icon="heart"
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders unchecked toggle button', () => {
  const tree = renderer
    .create(
      <ToggleButton
        disabled
        status="unchecked"
        onValueChange={() => {}}
        icon="heart"
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

describe('getToggleButtonColor', () => {
  const getTheme = (isDark = false, isV3 = true) => {
    const theme = isDark
      ? isV3
        ? MD3DarkTheme
        : MD2DarkTheme
      : isV3
      ? MD3LightTheme
      : MD2LightTheme;
    return {
      ...theme,
      isV3,
      md: (tokenKey) => get(theme.tokens, tokenKey),
    };
  };

  it('should return correct color when checked and theme version 3', () => {
    expect(getToggleButtonColor({ theme: getTheme(), checked: true })).toBe(
      color(getTheme().colors.onSecondaryContainer)
        .alpha(tokens.md.ref.opacity.level2)
        .rgb()
        .string()
    );
  });

  it('should return correct color when checked and theme version 3, dark theme', () => {
    expect(getToggleButtonColor({ theme: getTheme(true), checked: true })).toBe(
      color(getTheme(true).colors.onSecondaryContainer)
        .alpha(tokens.md.ref.opacity.level2)
        .rgb()
        .string()
    );
  });

  it('should return correct color when checked and theme version 2', () => {
    expect(
      getToggleButtonColor({ theme: getTheme(false, false), checked: true })
    ).toBe('rgba(0, 0, 0, .08)');
  });

  it('should return correct color when checked and theme version 2, dark theme', () => {
    expect(
      getToggleButtonColor({ theme: getTheme(true, false), checked: true })
    ).toBe('rgba(255, 255, 255, .12)');
  });

  it('should return transparent color when not checked', () => {
    expect(getToggleButtonColor({ theme: getTheme(), checked: false })).toBe(
      'transparent'
    );
  });
});
