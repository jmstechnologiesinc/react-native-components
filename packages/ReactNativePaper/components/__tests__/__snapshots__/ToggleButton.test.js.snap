// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders disabled toggle button 1`] = `
<View
  style={
    Object {
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0,
      "shadowRadius": 0,
    }
  }
>
  <View
    style={
      Object {
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(29, 25, 43, 0.12)",
          "borderColor": "rgba(121, 116, 126, 1)",
          "borderRadius": 4,
          "borderWidth": 0,
          "elevation": 0,
          "height": 42,
          "margin": 0,
          "overflow": "hidden",
          "width": 42,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": true,
            "selected": true,
          }
        }
        accessible={true}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            false,
            Object {
              "alignItems": "center",
              "flexGrow": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          accessibilityElementsHidden={true}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          selectable={false}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "color": "rgba(28, 27, 31, 0.38)",
                "fontSize": 24,
              },
            ]
          }
        >
          □
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders toggle button 1`] = `
<View
  style={
    Object {
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0,
      "shadowRadius": 0,
    }
  }
>
  <View
    style={
      Object {
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(29, 25, 43, 0.12)",
          "borderColor": "rgba(121, 116, 126, 1)",
          "borderRadius": 4,
          "borderWidth": 0,
          "elevation": 0,
          "height": 42,
          "margin": 0,
          "overflow": "hidden",
          "width": 42,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": undefined,
            "selected": true,
          }
        }
        accessible={true}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            false,
            Object {
              "alignItems": "center",
              "flexGrow": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          accessibilityElementsHidden={true}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          selectable={false}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "color": "rgba(73, 69, 79, 1)",
                "fontSize": 24,
              },
            ]
          }
        >
          □
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders unchecked toggle button 1`] = `
<View
  style={
    Object {
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0,
      "shadowRadius": 0,
    }
  }
>
  <View
    style={
      Object {
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "transparent",
          "borderColor": "rgba(121, 116, 126, 1)",
          "borderRadius": 4,
          "borderWidth": 0,
          "elevation": 0,
          "height": 42,
          "margin": 0,
          "overflow": "hidden",
          "width": 42,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": true,
            "selected": false,
          }
        }
        accessible={true}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            false,
            Object {
              "alignItems": "center",
              "flexGrow": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          accessibilityElementsHidden={true}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          selectable={false}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "color": "rgba(28, 27, 31, 0.38)",
                "fontSize": 24,
              },
            ]
          }
        >
          □
        </Text>
      </View>
    </View>
  </View>
</View>
`;
